SmellDetector/Detectors/RuleEngines/Rule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using SmellDetector.Controllers;
+using SmellDetector.SmellModel;
+
+namespace SmellDetector.Detectors.RuleEngines
+{
+    public class Rule
+    {
+        private readonly String _DOI;
+        private readonly Criteria _criteria;
+        private readonly SmellType _smellType;
+        public Rule(String DOI, Criteria criteria, SmellType smellType)
+        {
+            _DOI = DOI;
+            _criteria = criteria;
+            _smellType = smellType;
+        }
+
+        public Issue IsTriggered(String codeSnippetId, Dictionary<String, double> metrics)
```

Comment:
The "IsTriggered" name signals that a boolean should be returned. "Validate" might be more suitable.